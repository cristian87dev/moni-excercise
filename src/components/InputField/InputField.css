.input-field {
  color: var(--font-color);
  padding: 0 1rem;
  margin-bottom: 1rem;
}

.input-label, .input-value {
  display: block;
}

.input-label {

}

.input-value {
  padding: 0.8em;
  font-size: 1.3rem;
  border-radius: 3px;
  border: none;
  appearance: none;
  width: 100%;
}
