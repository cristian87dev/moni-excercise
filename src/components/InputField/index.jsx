import React from 'react';
import PropTypes from 'prop-types';

import './InputField.css';

const InputField = ({ label, value, name, type, placeholder, options, onChange }) => {
  const Label = <label className="input-label" htmlFor={name}>{label}</label>;
  let Input = null;

  if (!type || type === 'number' || type === 'text') {
    Input = <input
      className="input-value"
      type={type}
      id={name}
      name={name}
      placeholder={placeholder || name}
      value={value}
      autoComplete="off"
      onChange={onChange}
    />;
  } else if (type === 'select') {
    const Options = options.map(
      option => <option key={option.value} value={option.value || option.label}>
        {option.label}
      </option>
    );
    Input = <select
      className="input-value"
      id={name}
      name={name}
      placeholder={placeholder || name}
      value={value}
      autoComplete="off"
      onChange={onChange}>
      { Options }
    </select>;
  }

  return (
    <div className="input-field">
      { label && Label }
      { Input }
    </div>
  );
};

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ]),
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array
};

InputField.defaultProps = {
  type: 'text',
  value: '',
  placeholder: '',
  options: []
};

export default InputField;
