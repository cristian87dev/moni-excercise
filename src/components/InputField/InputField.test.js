import React from 'react';
import renderer from 'react-test-renderer';
import InputField from './index';

it('Basic-InputField renders without crashing', () => {
  const onChange = jest.fn();
  const component = renderer.create(
    <InputField name="test" onChange={onChange} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Select-InputField renders without crashing', () => {
  const onChange = jest.fn();
  const options = [{ name: 'name', value: 'value' }];
  const component = renderer.create(
    <InputField type="select" name="test" onChange={onChange} options={options} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
