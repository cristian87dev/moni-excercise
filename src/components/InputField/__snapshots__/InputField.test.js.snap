// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic-InputField renders without crashing 1`] = `
<div
  className="input-field"
>
  <input
    autoComplete="off"
    className="input-value"
    id="test"
    name="test"
    onChange={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
</div>
`;

exports[`Select-InputField renders without crashing 1`] = `
<div
  className="input-field"
>
  <select
    autoComplete="off"
    className="input-value"
    id="test"
    name="test"
    onChange={[Function]}
    placeholder="test"
    value=""
  >
    <option
      value="value"
    />
  </select>
</div>
`;
